src/syscheckd/run_check.c

```diff
@@ -120,6 +124,45 @@ void fim_send_scan_info(fim_scan_event event) {
     cJSON_Delete(json);
 }
 
+void check_max_fps(int baseline) {
+#ifndef WAZUH_UNIT_TESTING
+    static unsigned int files_read = 0;
+    static time_t last_time = 0;
+#endif
+    static pthread_mutex_t fps_mutex = PTHREAD_MUTEX_INITIALIZER;
+    static pthread_cond_t cond = PTHREAD_COND_INITIALIZER;
+    struct timespec wait_time = {0, 0};
+
+    if (syscheck.max_files_per_second == 0 || baseline == 0) {
```

Comment:
@antoniomanuelfr the use case we're approaching is to prevent the agent from consuming too much CPU or disk. 

IMHO this also applies to the first scan, we should apply this limit to the baseline, even if this delays the start of real-time and who-data.